src/core/project/qgsprojectstylesettings.cpp

```diff
@@ -0,0 +1,204 @@
+#include "qgslinesymbol.h"
+#include "qgsfillsymbol.h"
+#include "qgscolorramp.h"
+#include "qgstextformat.h"
+
+#include <QDomElement>
+
+QgsProjectStyleSettings::QgsProjectStyleSettings( QgsProject *project )
+  : QObject( project )
+  , mProject( project )
+{
+
+}
+
+QgsSymbol *QgsProjectStyleSettings::defaultSymbol( QgsWkbTypes::GeometryType geomType ) const
+{
+  switch ( geomType )
+  {
+    case QgsWkbTypes::PointGeometry:
+      return mDefaultMarkerSymbol ? mDefaultMarkerSymbol->clone() : nullptr;
+
+    case QgsWkbTypes::LineGeometry:
+      return mDefaultLineSymbol ? mDefaultLineSymbol->clone() : nullptr;
+
+    case QgsWkbTypes::PolygonGeometry:
+      return mDefaultFillSymbol ? mDefaultFillSymbol->clone() : nullptr;
+
+    default:
+      break;
+  }
+
+  return nullptr;
+}
+
+void QgsProjectStyleSettings::setDefaultSymbol( QgsWkbTypes::GeometryType geomType, QgsSymbol *symbol )
+{
+  switch ( geomType )
+  {
+    case QgsWkbTypes::PointGeometry:
+      mDefaultMarkerSymbol.reset( symbol ? symbol->clone() : nullptr );
+      break;
+
+    case QgsWkbTypes::LineGeometry:
+      mDefaultLineSymbol.reset( symbol ? symbol->clone() : nullptr );
+      break;
+
+    case QgsWkbTypes::PolygonGeometry:
+      mDefaultFillSymbol.reset( symbol ? symbol->clone() : nullptr );
+      break;
+
+    default:
+      break;
+  }
+}
+
+QgsColorRamp *QgsProjectStyleSettings::defaultColorRamp() const
+{
+  return mDefaultColorRamp ? mDefaultColorRamp->clone() : nullptr;
+}
+
+void QgsProjectStyleSettings::setDefaultColorRamp( QgsColorRamp *colorRamp )
+{
+  mDefaultColorRamp.reset( colorRamp ? colorRamp->clone() : nullptr );
+}
+
+QgsTextFormat QgsProjectStyleSettings::defaultTextFormat() const
+{
+  return mDefaultTextFormat;
+}
+
+void QgsProjectStyleSettings::setDefaultTextFormat( QgsTextFormat textFormat )
+{
+  mDefaultTextFormat = textFormat;
+}
+
+void QgsProjectStyleSettings::reset()
+{
+  mDefaultMarkerSymbol.reset();
+  mDefaultLineSymbol.reset();
+  mDefaultFillSymbol.reset();
+  mDefaultColorRamp.reset();
+  mDefaultTextFormat = QgsTextFormat();
+  mRandomizeDefaultSymbolColor = true;
+  mDefaultSymbolOpacity = 1.0;
+}
+
+bool QgsProjectStyleSettings::readXml( const QDomElement &element, const QgsReadWriteContext & )
+{
+  QgsReadWriteContext rwContext;
+  rwContext.setPathResolver( mProject->pathResolver() );
+
+  mRandomizeDefaultSymbolColor = element.attribute( QStringLiteral( "RandomizeDefaultSymbolColor" ), QStringLiteral( "0" ) ).toInt();
+  mDefaultSymbolOpacity = element.attribute( QStringLiteral( "DefaultSymbolOpacity" ), QStringLiteral( "1.0" ) ).toDouble();
+
+  QDomElement elem = element.firstChildElement( QStringLiteral( "markerSymbol" ) );
+  if ( !elem.isNull() )
+  {
+    QDomElement symbolElem = elem.firstChildElement( QStringLiteral( "symbol" ) );
+    if ( !symbolElem.isNull() )
+    {
+      mDefaultMarkerSymbol.reset( QgsSymbolLayerUtils::loadSymbol<QgsMarkerSymbol>( symbolElem, rwContext ) );
+    }
+  }
```

Comment:
Here the marker symbol should be reset -- when you're saving you aren't adding the element if the symbol is nullptr, so here a null element means the restored symbol should be nullptr too